pallets/rmrk-core/src/lib.rs

```diff
@@ -397,86 +378,19 @@ pub mod pallet {
 			let sender = match T::ProtocolOrigin::try_origin(origin) {
 				Ok(_) => None,
 				Err(origin) => Some(ensure_signed(origin)?),
-			};
-
-			let mut sending_nft =
-				NFTs::<T>::get(collection_id, nft_id).ok_or(Error::<T>::NoAvailableNftId)?;
-			ensure!(
-				sending_nft.rootowner == sender.clone().unwrap_or_default(),
-				Error::<T>::NoPermission
-			);
-
-			match new_owner.clone() {
-				AccountIdOrCollectionNftTuple::AccountId(account_id) => {
-					// Remove previous parental relationship
-					if let AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) =
-						sending_nft.owner
-					{
-						if let Some(mut kids) = Children::<T>::take(cid, nid) {
-							kids.retain(|&kid| kid != (collection_id, nft_id));
-							Children::<T>::insert(cid, nid, kids);
-						}
-					}
-					sending_nft.rootowner = account_id.clone();
-
-					// Pallet::<T>::recursive_update_rootowner(
-					// 	collection_id,
-					// 	nft_id,
-					// 	account_id.clone(),
-					// 	T::MaxRecursions::get(),
-					// )?;
-				},
-				AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) => {
-					let recipient_nft =
-						NFTs::<T>::get(cid, nid).ok_or(Error::<T>::NoAvailableNftId)?;
-					// Check if sending NFT is already a child of recipient NFT
-					ensure!(
-						!Pallet::<T>::is_x_descendent_of_y(cid, nid, collection_id, nft_id),
-						Error::<T>::CannotSendToDescendent
-					);
-
-					// Remove parent if exists: first we only care if the owner is a non-AccountId)
-					if let AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) =
-						sending_nft.owner
-					{
-						// second we only care if the parent has children (it should)
-						if let Some(mut kids) = Children::<T>::take(cid, nid) {
-							// third we only "retain" the other children
-							kids.retain(|&kid| kid != (collection_id, nft_id));
-							Children::<T>::insert(cid, nid, kids);
-						}
-					}
-					if sending_nft.rootowner != recipient_nft.rootowner {
-						// sending_nft.rootowner = recipient_nft.rootowner
-						sending_nft.rootowner = recipient_nft.rootowner.clone();
-
-						Pallet::<T>::recursive_update_rootowner(
-							collection_id,
-							nft_id,
-							recipient_nft.rootowner,
-							T::MaxRecursions::get(),
-						)?;
-					}
-					match Children::<T>::take(cid, nid) {
-						None => Children::<T>::insert(cid, nid, vec![(collection_id, nft_id)]),
-						Some(mut kids) => {
-							kids.push((collection_id, nft_id));
-							Children::<T>::insert(cid, nid, kids);
-						},
-					}
-				},
-			};
-			sending_nft.owner = new_owner.clone();
-
-			NFTs::<T>::remove(collection_id, nft_id);
-			NFTs::<T>::insert(collection_id, nft_id, sending_nft);
+			}
+			.unwrap_or_default();
 
-			Self::deposit_event(Event::NFTSent(
-				sender.unwrap_or_default(),
-				new_owner,
+			let max_recursions = T::MaxRecursions::get();
+			<Self as Nft<T::AccountId, StringLimitOf<T>>>::send(
```

Comment:
I think if you have `use Nft`, you can just write `Self::send()`. Not 100% sure, but we can have a try

Reply:
Indeed, just need to rename `send` on Nft trait otherwise naming collision with pallet call.

Reply:
Addressed in https://github.com/rmrk-team/rmrk-substrate/pull/28/commits/4c1a84f45590d5ede7d71e4440bdd99669ae7285